usl_pipeline/usl_lib/usl_lib/readers/polygon_readers.py

```diff
@@ -0,0 +1,49 @@
+import typing
+from typing import Tuple
+
+from shapely import geometry
+
+
+def read_polygons_from_text_file(
+    file: typing.TextIO, support_mask_values: bool = False
+) -> list[Tuple[geometry.Polygon, int]]:
+    """Reads polygon information with optional mask values from text file.
+
+    Args:
+        file: File text stream to load from.
+        support_mask_values: Optional indicator of the presence of polygon mask values
+            in the first column in polygon lines. Default case is that mask column is
+            not present (in this case mask value 1 will be assigned to every polygon).
+
+    Returns:
+        The list of tuples combining polygon with associated mask.
+    """
+    lines = file.readlines()
```

Comment:
That's going to read the entire file into memory at once.
Can we process this file line by line instead?

Reply:
There is no need because shape files are relatively small and you'll get all the polygon objects in memory anyway.